sdk/core/core/test/cmocka/test_az_span.c

```diff
@@ -760,6 +761,54 @@ static void az_span_copy_empty(void** state)
   assert_true(az_span_is_content_equal(az_span_copy(dst, AZ_SPAN_NULL), dst));
 }
 
+static void test_az_span_is_valid(void** state)
+{
+  (void)state;
+  {
+    az_span const span_zeroed = { 0 };
+    az_span const span_null = AZ_SPAN_NULL;
+    assert_true(memcmp(&span_zeroed, &span_null, sizeof(az_span)) == 0);
+  }
+
+  assert_true(_az_span_is_valid(AZ_SPAN_NULL, 0, true));
+  assert_false(_az_span_is_valid(AZ_SPAN_NULL, 0, false));
+  assert_false(_az_span_is_valid(AZ_SPAN_NULL, 1, true));
+  assert_false(_az_span_is_valid(AZ_SPAN_NULL, 1, false));
+  assert_false(_az_span_is_valid(AZ_SPAN_NULL, -1, true));
+  assert_false(_az_span_is_valid(AZ_SPAN_NULL, -1, false));
+
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR(""), 0, true));
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR(""), 0, false));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR(""), 1, true));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR(""), 1, false));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR(""), -1, true));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR(""), -1, false));
+
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 0, true));
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 0, false));
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 1, true));
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 1, false));
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 5, true));
+  assert_true(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 5, false));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 6, true));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), 6, false));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), -1, true));
+  assert_false(_az_span_is_valid(AZ_SPAN_FROM_STR("Hello"), -1, false));
+
+  uint8_t* const max_ptr = (uint8_t*)~0;
+  assert_true(_az_span_is_valid(az_span_init(max_ptr, 0), 0, false));
+  assert_true(_az_span_is_valid(az_span_init(max_ptr, 0), 0, true));
+
+  assert_false(_az_span_is_valid(az_span_init(max_ptr, 1), 0, false));
+  assert_false(_az_span_is_valid(az_span_init(max_ptr, 1), 0, true));
+
+  assert_true(_az_span_is_valid(az_span_init(max_ptr - 1 , 1), 0, false));
+  assert_true(_az_span_is_valid(az_span_init(max_ptr - 1, 1), 0, true));
+
+  assert_false(_az_span_is_valid(az_span_init(max_ptr - 1, 2), 0, false));
+  assert_false(_az_span_is_valid(az_span_init(max_ptr - 1, 2), 0, true));
```

Comment:
How about also for `max_size` (i.e. INT_MAX).
```C
assert_false(_az_span_is_valid(az_span_init(max_ptr - 1, max_size), 0, false));
assert_false(_az_span_is_valid(az_span_init(max_ptr - 1, max_size), 0, true));

assert_true(_az_span_is_valid(az_span_init(max_ptr - max_size, max_size), 0, false));
assert_true(_az_span_is_valid(az_span_init(max_ptr - max_size, max_size), 0, true));
```

Reply:
Done.